Slim/Player/Protocols/HTTP.pm

```diff
@@ -368,6 +447,133 @@ sub canDirectStreamSong {
 	return $direct;
 }
 
+sub readChunk {
+	my $self  = $_[0];
+	my $v = ${*$self}{'_enhanced'} || return $self->_sysread($_[1], $_[2], $_[3]);
+	return $v->{'reader'}->($v, $self, $_[1], $_[2], $_[3]);
```

Comment:
Shouldn't this be `$v->{'_enhanced'}->{'reader'}`?

Reply:
No, $v (that you like) is already a hash to '_enhanced'

Reply:
Heh... there's the reason why a better name might have helped ;-)

Reply:
Done